modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,153 @@
 
 import {submodule} from '../src/hook.js'
 import { getStorageManager } from '../src/storageManager.js';
+import { triggerPixel, logInfo } from '../src/utils.js';
+import { uspDataHandler, coppaDataHandler, gdprDataHandler } from '../src/adapterManager.js';
 
 const QUANTCAST_FPA = '__qca';
+const DEFAULT_COOKIE_EXP_TIME = 392; // (13 months - 2 days)
+const PREBID_PCODE = 'p-KceJUEvXN48CE'; // Not associated with a real account
+const DOMAIN_QSERVE = 'https://pixel.quantserve.com/pixel';
+const QUANTCAST_VENDOR_ID = '11';
+const PURPOSE_DATA_COLLECT = '1';
+const PURPOSE_PRODUCT_IMPROVEMENT = '10';
+const QC_TCF_REQUIRED_PURPOSES = [PURPOSE_DATA_COLLECT, PURPOSE_PRODUCT_IMPROVEMENT];
+const QC_TCF_CONSENT_FIRST_PURPOSES = [PURPOSE_DATA_COLLECT];
+const QC_TCF_CONSENT_ONLY_PUPROSES = [PURPOSE_DATA_COLLECT];
+const GDPR_PRIVACY_STRING = gdprDataHandler.getConsentData();
+const US_PRIVACY_STRING = uspDataHandler.getConsentData();
 
 export const storage = getStorageManager();
 
+export function firePixel(clientId, cookieExpTime = DEFAULT_COOKIE_EXP_TIME) {
+  // check for presence of Quantcast Measure tag _qevent obj,
+  // required gdpr consent / legitimate interest for purpose 1 and 10
+  // for Quantcast as a vendor, and publisher provided clientID
+  if (!window._qevents && hasGDPRConsent(GDPR_PRIVACY_STRING) && clientId && clientId != '') {
+    var fpa = storage.getCookie(QUANTCAST_FPA);
+    var fpan = '0';
+    var domain = quantcastIdSubmodule.findRootDomain();
+    var now = new Date();
+    var usPrivacyParamString = '';
+    var firstPartyParamStrings;
+    var gdprParamStrings;
+
+    if (!fpa) {
+      var expires = new Date(now.getTime() + (cookieExpTime * 86400000)).toGMTString();
+      fpa = 'B0-' + Math.round(Math.random() * 2147483647) + '-' + et;
```

Comment:
I like that we're following the pattern of using a different alnum prefix for each type of fpa. :+1: 

Where is `et` defined? I don't see it here and I don't see it in the prebid codebase.

```
> git grep '\<et\>' **/*.js| grep -v test
modules/quantcastIdSystem.js:      fpa = 'B0-' + Math.round(Math.random() * 2147483647) + '-' + et;
```
It seems to not be defined even in this branch.

Reply:
Corrected it. I had accidentally removed it in previous commit